MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -101,59 +43,80 @@ final class MovieQuizViewController: UIViewController {
             imageView.layer.borderColor = UIColor.ypRed.cgColor
         }
         
-        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) { [weak self] in
+            guard let self = self else { return }
             self.showNextQuestionOrRoundResults()
+            sender.isEnabled = true
         }
     }
     
-    private func showRoundResult(model: QuizResultsViewModel) {
-        let alert = UIAlertController(title: model.title,
-                                      message: model.text,
-                                      preferredStyle: .alert)
-        let action = UIAlertAction(title: model.buttonText, style: .default) { _ in
-            self.currentQuestionIndex = 0
-            self.correctAnswers = 0
-            let viewModel = self.convert(model: questions[self.currentQuestionIndex])
-            self.showNextQuestion(quiz: viewModel)
-        }
-        alert.addAction(action)
-        self.present(alert, animated: true)
-    }
-    
     private func showNextQuestionOrRoundResults() {
         imageView.layer.borderWidth = 0
-        if currentQuestionIndex == questions.count - 1 {
-            let viewModel = QuizResultsViewModel(
-                title: "Этот раунд окончен!",
-                text: "Ваш результат: \(correctAnswers)/10",
-                buttonText: "Сыграть ещё раз")
-            showRoundResult(model: viewModel)
+        if currentQuestionIndex == questionAmount - 1 {
+            guard let statisticService = statisticService else { return }
+            statisticService.store(correct: correctAnswers, total: questionAmount)
+            let date = statisticService.bestGame.date.dateTimeString
+            let message = """
+                Ваш результат: \(correctAnswers)/\(questionAmount)\n\
+                Количесство сыгранных квизов: \(statisticService.gamesCount)
+                Рекорд: \(statisticService.bestGame.correct)/\(statisticService.bestGame.total) (\(date))
+                Средняя точность: \(statisticService.totalAccuracy)
+            """
+            let alertModel = AlertModel(title: "Этот раунд окончен!",
+                                        message: message,
+                                        buttonText: "Сыграть ещё раз",
+                                        complection: { [weak self] in
+                                            guard let self = self else { return }
+                                            self.currentQuestionIndex = 0
+                                            self.correctAnswers = 0
+                                            self.questionFactory?.requestNextQuestion()
+                                        })
```

Comment:
я не очень люблю такой стиль-столбиком, потому что каждый вложенный уровень делает очень большой сдвиг вправо, отдаю предпочтение такому стилю, хотя и твой имеет право на жизнь:
```suggestion
            let alertModel = AlertModel(
                title: "Этот раунд окончен!",
                message: message,
                buttonText: "Сыграть ещё раз",
                completion: { [weak self] in
                    guard let self = self else { return }
                    self.currentQuestionIndex = 0
                    self.correctAnswers = 0
                    self.questionFactory?.requestNextQuestion()
                })
```

кстати обрати внимание на опечатку в слове completion(лучше включить подсветку для опечаток, https://sarunw.com/posts/spell-checking-in-xcode/)